Added support for Subresource Integrity

thejoeejoee opened the pull request:
* added support for SRI via `INTEGRITY` configuration option
* `integrity: true` configuration for the bundle tracker is needed

fjsj commented on webpack_loader/loader.py:
```diff
@@ -53,9 +65,15 @@ def map_chunk_files_to_url(self, chunks):
         assets = self.get_assets()
         files = assets['assets']
 
+        add_integrity = self.config['INTEGRITY']
```
To keep backwards compatibility, I think this should be `self.config.get('INTEGRITY', False)`

fjsj commented on webpack_loader/utils.py:
```diff
@@ -68,12 +71,21 @@ def get_as_tags(bundle_name, extension=None, config='DEFAULT', suffix='', attrs=
+                ).format(
+                    ''.join([chunk['url'], suffix]),
+                    attrs,
+                    loader.get_integrity_attr(chunk) or ' ',
```
do we need ` or ' ',`?

fjsj commented on tests/app/tests/test_webpack.py:
```diff
@@ -210,6 +210,34 @@ def test_preload(self):
+        self.compile_bundles('webpack.config.integrity.js')
+
+        loader = get_loader(DEFAULT_CONFIG)
+        with patch.dict(loader.config, {'INTEGRITY': True}):
```
Please add another test that uses a `loader.config` without any `'INTEGRITY'` key. It should work as `'INTEGRITY': False`. This will ensure backwards compatibility.

thejoeejoee replied:
Added `test_integrity_missing_config`:

https://github.com/django-webpack/django-webpack-loader/pull/315/commits/e880336976ae77f750db2d38707c438411492e46#diff-1af3f1863bb5f5c3e50ccb014a6b83fd14ab4fe4dbc268c6285f119bbb9885b0R232-R254

fjsj reviewed (changes requested):
Thanks, PR looks great, but I think we need some changes to keep backwards compatibility on integrity config.

thejoeejoee commented:
glad to contribute:

* added test case for missing `INTEGRITY` config key
* replaced direct config indexing by `dict.get()`
* moved whitespace attr separator to method constructing the attribute itself

fjsj commented:
Thanks @thejoeejoee . I made a small fix at https://github.com/django-webpack/django-webpack-loader/pull/315/commits/6a5a880215d1481772e4594b71153cd1548df98f and merged.
